src/settings/panel.tsx

```diff
@@ -510,6 +689,7 @@ export class AiSettings extends React.Component<
     );
   }
 
+  private _usage: modelUsage;
```

Comment:
Perhaps we could improve the name of the choice variable between `chat` and `completion`.
Currently, the type is called `modelUsage` and the variable `_usage` (in several places), I haven't found anything more relevant.

Reply:
I meant having both

```ts
get currentChatName(): string 
```

and

```ts
get currentCompleterName(): string
```

as an alternative to:

```ts
currentName(usage: ModelUsage): string
```

If that can help avoid having to expose a `ModelUsage` type.

Reply:
Not sure to get it.
AFAIU, what you suggest would probably be enough for the provider registry, but we would have to duplicate also the functions that save settings in the registry and in local storage (https://github.com/jupyterlite/ai/pull/105/files#diff-c163aac3bfa8d91d61fc51c382f75f20f4aa46ef650e860db9d52b8a65a3670aR52).
Do you think that it would be a better implementation ?

Reply:
I updated it (and rebased from main)